File: competition/08-collections/collections.js
const nums = [];
for (let i = 1; i <= 1000000; i++) nums.push(i);
const squared = nums.map((x) => x * x);
const evens = squared.filter((x) => x % 2 === 0);
const total = evens.reduce((acc, x) => (acc + x) % 1000000007, 0);
console.log(total);

File: competition/08-collections/collections.rb
nums = (1..1000000).to_a
squared = nums.map { |x| x * x }
evens = squared.select { |x| x % 2 == 0 }
total = evens.reduce(0) { |acc, x| (acc + x) % 1000000007 }
puts total
